### ISSUE TITLE
What is the right way to handle user session?

### ISSUE THREAD

**Issue by wilburx9** [amplitude/experiment-android-client#40]
We run an experiment before the user log in and after they log in. We initialize the client like so:

```
val config = ExperimentConfig.builder()
      .build()
val experiment = Experiment.initializeWithAmplitudeAnalytics(app, $API_KEY  config,)
val user = ExperimentUser.builder()
      .userId(userAuth0Id)
      .deviceId(deviceIdFromAnalytics)
      .build()
experiment?.start(user)?.get()
```
Before the user logs in, `userAuth0Id` is `null` and calling `experiment?.variant(variantKey)` returns the expected variant.

After logging in, we call the code above with a valid `userAuth0Id`. But the now logged-in user doesn't get the expected variant of the experiment that is assigned to their `userId`, the properties in the object returned in `experiment?.variant(variantKey)` are just `null`. I have tried calling `experiment?.clear()` after login but that doesn't fix it.

However, if you restart the app, they get the right values.

What is the right way to handle this?

I am using v1.11.0

**Comment by tyiuhc** [comment]
Hi @wilburx9, many thanks for reaching out. In your implementation, after the user logs in, do you re-initialize the client as well, or just call `experiment?.variant(variantKey)` again? If it's the latter, you should use `setUser()` on the new `user` object (with valid `userAuth0Id`) before calling `variant()`. Let us know if this helps with your issue.

**Comment by wilburx9** [comment]
After the user logs in, I call:

```
val config = ExperimentConfig.builder()
      .build()
val experiment = Experiment.initializeWithAmplitudeAnalytics(app, $API_KEY  config,)
val user = ExperimentUser.builder()
      .userId(userAuth0Id)
      .deviceId(deviceIdFromAnalytics)
      .build()
experiment?.start(user)?.get()
```
And `userAuth0Id` is a valid non-null and non-empty string at this point.

**Comment by tyiuhc** [comment]
This issue has been addressed in v1.11.1.
